### DISCUSSION TITLE
rspec-puppet renaming

### DISCUSSION THREAD

**Discussion by davidsandilands** [puppetlabs/rspec-puppet#89] [General]
The rspec-puppet gem published in https://rubygems.org/gems/rspec-puppet currently has shared ownership with the original author. To ensure good contribution and maintenance standards we would propose republishing this gem into a new namespace.

Following initial discussions we did for puppet-lint https://github.com/puppetlabs/puppet-lint/pull/158 we found issues with namespace clashes following an initial renaming to https://rubygems.org/gems/puppetlabs-puppet-lint which was investigated in https://github.com/puppetlabs/puppet-lint/pull/170

I am therefore opening this issue to discuss this need fully and address any community concerns before proceeding.

So with that I'd like to lay out what were suggesting now and make sure we aren't missing anything further.

To avoid a clash anywhere, gem, namespace etc and to drive consistency of repos we are planning to rename the gem, the binary and the repo to follow puppetlabs- , we thought dropping the "puppet" part to avoid a long name and keep consistency with the forge and associations of puppetlabs with puppet inc and puppet with the community.

Our current proposal would be to rename accross binary, repo and gem as follows

Repo rename: puppetlabs-rspec
Binary name: puppetlabs-rspec
Gem rename: puppetlabs-rspec

Thoughts? Concerns? Feedback?

See related discussion https://github.com/puppetlabs/puppet-lint/discussions/173

**Comment by genebean** [comment]
The standard for all plugins to rspec is for the gem to be `rspec-<plugin name>`. You can see plenty of precedent for this via [https://rubygems.org/search?query=rspec-](https://rubygems.org/search?query=rspec-).

**Reply by ekohl** [reply]
The `rspec-<plugin>` naming is very common to identify it's an rspec plugin. There is [beaker-rspec](https://github.com/voxpupuli/beaker-rspec) and you could argue that it should be `rspec-beaker` instead. I'd prefer to keep the `rspec-` prefix.

Perhaps a weird suggestion, but is `rspec-puppet2` an option?

**Reply by yachub** [reply]
> Perhaps a weird suggestion, but is rspec-puppet2 an option?

FWIW, here's a random example of that pattern used with another popular gem https://github.com/ManageIQ/rbvmomi2

**Reply by davidsandilands** [reply]
Linking Genes comment https://github.com/puppetlabs/rspec-puppet/discussions/89#discussioncomment-7403241 I would agree that a 2 can make it seemed versioned as opposed to essentially forked in this case.

The original thought had been puppetlabs-rspec-puppet which felt unnecessarily long and that keeping puppetlabs first kept our gem naming consistent.

**Reply by ekohl** [reply]
I'd go with `rspec-puppetlabs-puppet` since the rspec ecosystem is more important that Puppetlabs' naming.

**Comment by genebean** [comment]
Regarding ownership, I addressed that in https://github.com/puppetlabs/puppet-lint/discussions/173#discussioncomment-7402792 and it seems relevant here too.

**Reply by davidsandilands** [reply]
Just to give the summary the quick summary it is a social thing and its therefore not transferrable.

**Comment by genebean** [comment]
The issue with a number is it could easily be mistaken as one correlating to the puppet version number.

**Comment by davidsandilands** [comment]
So I want to summarize the points taken on each of the discussions and our future actions.

We are aware this change is frustrating for everyone and these sort of rename changes have risks but this wasn't a project we could transfer or where there was an active partnership we could make to justify the joint ownership.

We are using the puppetlabs-<toolname> approach since that maintains some level of consistency with the forge. We appreciate quite a few rspec gems follow an rspec-<toolname> approach but there are a number which don't and we consider it to be better to maintain consistent naming over the tools.

We believe renaming the binary is the right thing to do to avoid potential clashes and the confusion of having an old and new gem but with the same binary name. Although we are aware historically binaries were more likely to use puppet with this generally masked by use of the pdk and rake this should hopefully be a minimal disruption and inconsistency

We are aware we could have started these discussions earlier and have updated our ticket grooming process to reflect that. We will also be raising PRs into dependencies to try and reduce work for  the community making the change.

I appreciate not everyone will be entirely happy with the naming outcome and potential disruption but we hope applying a consistent approach will reduce disruption in future. I really appreciate everys input on this to help us see all available options.

**Comment by davidsandilands** [comment]
Reopening

As per discussion in Puppet-lint @gavindidrichsen @jordanbreen28 and co have been doing a hack day today to test the idea discussed on the vox meeting of releasing a gem in github and packing that in PDK so that we can maintain a secure pipeline for PDK packaging while at the same time still publishing to rubygems with the understanding of the ownership there. 

We will come back with findings of hack day and confirm direction

**Comment by jordanbreen28** [comment]
see https://github.com/puppetlabs/puppet-lint/discussions/173#discussioncomment-7643642
